content/automate/ManageIQ/Transformation/TransformationHosts/Common.class/__methods__/utils.rb

```diff
@@ -0,0 +1,66 @@
+#
+# 
+#
+
+module Transformation
+  module TransformationHost
+    module Common
+      class Utils
+        def initialize(handle = $evm)
+          @debug = true
+          @handle = handle
+        end
+
+        def get_transformation_hosts(ems)
+          ems.hosts.select { |host| host.tagged_with('mf_transformation_host', 'true') }
+        end
+        
+        def get_runners_count_by_host(host)
+          return @handle.vmdb(:vm).all.select { |v| v.custom_get('Transformation Host') == host.name }.size
+        end
+        
+        def get_runners_count_by_ems(ems)
+          total_runners = 0
+          get_transformation_hosts.each { |host| total_runners += get_runners_count_by_host(host) }
+          return total_runners
+        end
+
+        def set_transformation_host(vm, ems, factory_config)
+          # Determine the least utilized conversion host
+          min_runners, transformation_host = nil, nil
+          
+          get_transformation_hosts.each do |host|
+            runners_count = get_runners_count_by_host(host)
+            max_runners = host.custom_get('Max Transformation Runners')
+            max_runners = factory_config['transformation_host_max_runners'] if max_runners.blank?
+            next if runners_count == max_runners
+            ( transformation_host = host ; break ) if runners_count == 0
+            if transformation_host.nil? || runners_count < min_runners
+              transformation_host = host
+              min_runners = runners_count
+            end
+          end
+          if transformation_host.nil?
+            @handle.log(:info, "No transformation host found.")
+          else
+            @handle.log(:info, "Least busy transformation host: #{transformation_host}")
+            vm.custom_set('Transformation Host', transformation_host.guid)
+          end
+          return transformation_host
+        end
+        
+        def unset_transformation_host(vm)
+          vm.custom_set('Transformation Host', nil)
```

Comment:
@fdupont-redhat can the custom attribute be deleted instead of being set to nil. Will the nil logic be used elsewhere to compare completed transformations.

Reply:
This is how you do it.  There are only three methods exposed `custom_keys, custom_get` and `custom_set`.

See [/app/models/mixins/custom_attribute_mixin.rb#L82](https://github.com/ManageIQ/manageiq/blob/d4a91f5434110a5be2c2cbdddce40349626e6bf1/app/models/mixins/custom_attribute_mixin.rb#L82)

Reply:
It will be changed, as the transformation host will be part of the options hash of the transformation task, rather that the VM.